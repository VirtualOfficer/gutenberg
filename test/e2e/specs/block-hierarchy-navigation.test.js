/**
 * Internal dependencies
 */
import {
	newPost,
	insertBlock,
	getEditedPostContent,
	pressTimes,
} from '../support/utils';

describe( 'Navigating the block hierarchy', () => {
	beforeAll( async () => {
		await newPost();
	} );

	it( 'Should navigate using the block hierarchy dropdown menu', async () => {
		await insertBlock( 'Columns (beta)' );

		// Add a paragraph in the first column
		await page.click( '.editor-default-block-appender__content' );
		await page.keyboard.type( 'First column' );

		// Navigate to the columns blocks
		await page.click( '[aria-label="Open Block Hierarchy Navigator"]' );
		const columnsBlockMenuItem = ( await page.$x( "//button[contains(@class,'editor-block-hierarchy__item') and contains(text(), 'core/columns')]" ) )[ 0 ];
		await columnsBlockMenuItem.click();

		// Tweak the columns count
		await page.focus( '.edit-post-block-sidebar__panel .components-range-control__number[aria-label="Columns"]' );
		page.keyboard.press( 'Backspace' );
		page.keyboard.type( '3' );

		// Navigate to the last column blosck
		await page.click( '[aria-label="Open Block Hierarchy Navigator"]' );
		const lastColumnsBlockMenuItem = ( await page.$x( "//button[contains(@class,'editor-block-hierarchy__item') and contains(text(), 'core/column')]" ) )[ 3 ];
		await lastColumnsBlockMenuItem.click();

		// Insert text in the last column block
		await pressTimes( 'Tab', 2 ); // Navigate to the appender
		await page.keyboard.type( 'Third column' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
