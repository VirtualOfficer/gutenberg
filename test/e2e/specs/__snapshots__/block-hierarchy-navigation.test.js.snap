// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigating the block hierarchy Should navigate using the block hierarchy dropdown menu 1`] = `
"<!-- wp:columns {\\"columns\\":3} -->
<div class=\\"wp-block-columns has-3-columns\\"><!-- wp:column -->
<div class=\\"wp-block-column\\"></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class=\\"wp-block-column\\"></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class=\\"wp-block-column\\"><!-- wp:paragraph -->
<p>Third column</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->"
`;
